core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -157,37 +163,46 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def qualifyApp(
       path: EventLogInfo,
       hadoopConf: Configuration): Unit = {
+    var message: String = ""
     try {
       val startTime = System.currentTimeMillis()
       val app = QualificationAppInfo.createApp(path, hadoopConf, pluginTypeChecker, reportSqlLevel,
         mlOpsEnabled)
       if (!app.isDefined) {
         progressBar.foreach(_.reportUnkownStatusProcess())
-        logWarning(s"No Application found that contain SQL for ${path.eventLog.toString}!")
-        None
+        message = s"No Application found that contain SQL for ${path.eventLog.toString}!"
+        logWarning(message)
+        appStatusMap.put(path, StatusNotAvailable(None, message))
       } else {
         val qualSumInfo = app.get.aggregateStats()
         if (qualSumInfo.isDefined) {
           allApps.add(qualSumInfo.get)
+          appStatusMap.put(path, StatusSuccess(app))
           progressBar.foreach(_.reportSuccessfulProcess())
           val endTime = System.currentTimeMillis()
           logInfo(s"Took ${endTime - startTime}ms to process ${path.eventLog.toString}")
         } else {
           progressBar.foreach(_.reportUnkownStatusProcess())
-          logWarning(s"No aggregated stats for event log at: ${path.eventLog.toString}")
+          message = s"No aggregated stats for event log at: ${path.eventLog.toString}"
+          logWarning(message)
+          appStatusMap.put(path, StatusNotAvailable(app, message))
```

Comment:
Same comment as above. We should not keep a reference to `app` so that GC can free-up the memory for further allocation.

Reply:
Removed storing the entire app. Storing just the `appId` string